src/sema/ast.rs

```diff
@@ -1695,7 +1695,7 @@ pub struct TryCatch {
     pub errors: Vec<(Option<usize>, Parameter, Vec<Statement>)>,
     pub catch_param: Option<Parameter>,
     pub catch_param_pos: Option<usize>,
-    pub catch_stmt: Vec<Statement>,
+    pub catch_stmt: Option<Vec<Statement>>,
```

Comment:
So we have three options:
1. No `catch (e) { }` statement: all options are `None`
2. catch without parameter `catch { }`: `catch_stmt`: `Some` and `catch_param(_pos)`: `None`
3. catch with parameter: `catch (bytes x) {}`: all `Some`

Maybe it would make more sense to have:
```rust
pub struct CatchCase {
    pub param: Option<Parameter>,
    pub param_pos: Option<usize>,
    pub stmt: Vec<Statement>,
}
```
and then `struct TryCatch` can contain:
```rust
      pub cases: Vec<CatchCase>,
      pub catch_all: Option<CatchCase>,
```

Reply:
Yeah we could do something like this, it's more intuitive. I changed it